src/main/java/mekanism/common/util/MekanismUtils.java

```diff
@@ -767,18 +767,23 @@ public static Map<Fluid, FluidInDetails> getFluidsIn(Player player, UnaryOperato
     }
 
     public static void veinMineArea(IEnergyContainer energyContainer, Level world, BlockPos pos, ServerPlayer player, ItemStack stack, Item usedTool,
-          Collection<BlockPos> found, Function<Float, FloatingLong> destroyEnergyFunction, DoubleUnaryOperator distanceMultiplier, BlockState sourceState) {
+          Map<BlockPos, BlockState> found, Function<Float, FloatingLong> destroyEnergyFunction, BiFunction<Double, BlockState, Double> distanceMultiplier, BlockState sourceState) {
```

Comment:
As the cache isn't exactly useful here and as discussed I believe may function improperly at times I think this should be switched back to just being a collection and then the map's `keySet` can be passed to it.